h3pandas/h3pandas.py

```diff
@@ -758,6 +758,61 @@ def polyfill_resample(
 
         return result.h3.h3_to_geo_boundary() if return_geometry else result
 
+    def linetrace(
+        self, resolution : int, explode: bool = False, index_parts: bool = False
+    ) -> AnyDataFrame:
+        """Experimental. An H3 cell representation of a (Multi)LineString,
+        which permits repeated cells, but not if they are repeated in
+        immediate sequence.
+
+        Parameters
+        ----------
+        resolution : int
+            H3 resolution
+        explode : bool
+            If True, will explode the resulting list vertically.
+            All other columns' values are copied.
+            Default: False
+        index_parts : bool
+            If True, for the resulting index will be a multi-index (original
```

Comment:
I found this parameter hard to understand. In `GeoDataFrame.explode` it makes sense, as it's a parameter changing how `explode` behaves, but here I had to read to code to understand that it causes `explode` to be called.

What do you think about renaming it to something like `explode_geometry`, explaining that the function explodes the geometry prior to applying linetrace? Possibly also providing an `Example` for it.

Alternatively, we could argue the user can always explode first on their side and we don't need to provide this convenience parameter in H3-Pandas. 

Happy to hear your thoughts since you probably have the motivating usecase here and thought about this more deeply :)

Reply:
No you're right; I'm not sure what my line of thinking was there. Something about the use of multiindexes to handle multilinestring geometries. All it does is `explode` the GeoDataframe before doing anything else, which a user can do independently. I've removed it and adapted the relevant tests to do this explode independently.